Odata-docs/odatalib/whats-new/odatalib-8.md

```diff
@@ -0,0 +1,429 @@
+        "details": [
+            {
+                "code": "OEDC1",
+                "target": "OEDT1",
+                "message": "OEDM1"
+            },
+            {
+                "code": "OEDC1",
+                "message": "OEDM1"
+            },
+            {
+                "code": "",
+                "message": ""
+            }
+        ]
+    }
+}
+```
+
+Here's how it would be written in OData .NET 7 with `odataError.ToString()`:
+
+```json
+{
+  "error": {
+    "code": "",
+    "message": "",
+    "target": "",
+    "details": [
+      {
+        "errorcode": "OEDC1",
+        "message": "OEDM1",
+        "target": "OEDT1"
+      },
+      {
+        "errorcode": "OEDC1",
+        "message": "OEDM1",
+        "target": ""
+      },
+      {
+        "errorcode": "",
+        "message": "",
+        "target": ""
+      }
+    ],
+    "innererror": {}
+  }
+}
+```
+
+Here's how it would be serialized in OData .NET 8 with either `ODataJsonWriterUtils.WriteError(odataError, /* other parameters */)` and `odataError.ToString()`:
+
+```json
+{
+  "error": {
+    "code": "",
+    "message": "",
+    "details": [
+      {
+        "code": "OEDC1",
+        "message": "OEDM1",
+        "target": "OEDT1"
+      },
+      {
+        "code": "OEDC1",
+        "message": "OEDM1"
+      },
+      {
+        "code": "",
+        "message": ""
+      }
+    ]
+  }
+}
+```
+
+### Changed `ODataLibraryCompatibility` into a flags enum
+
+OData .NET 8 changed the `ODataLibraryCompatibility` enum into a flags enum, where each bit represents a different compatibility setting that can enable some legacy serialization behavior.
+
+In OData .NET 7, the enum had values like `ODataLibraryCompatibility.Version6`, `Version7` and `Latest`. These values were used to enable or disable certain behaviour based on desired compatibility with a certain library version after migration to a new version. In OData .NET 8, the we moved to a more granular option where you can set various compatibility options individually using bit flags, for example `ODataLibraryCompatibility.UseLegacyVariableCasing`, `WriteTopLevelODataNullAnnotation`, etc. This allows you to retain some legacy behaviour in the serializers when migrating to new versions. The enum still has `Version6` and `Version7` values, which now combine specific flags that are replicate the behaviour that was on by default for that specific version.
+
+### Serialize `variable` scale in lowercase
+
+When writing the `Scale` attribute in XML CSDL, use `variable` in lowercase instead of `Variable`. An enum flag, `ODataLibraryCompatibility.UseLegacyVariableCasing`, was added to support the legacy behavior if desired.
+
+### Changed JSON reader and writer interfaces
+
+This is relevant if you implemented custom JSON reader or JSON writer interfaces.
+
+In OData .NET 7, there were many interfaces required to implement a custom JSON reader or write because the synchronous, asynchronous and streaming APIs were split.
```

Comment:
```suggestion
In OData .NET 7, there were many interfaces required to implement a custom JSON reader or writer because the synchronous, asynchronous and streaming APIs were split.
```